android-components/components/feature/session/src/main/java/mozilla/components/feature/session/FullScreenFeature.kt

```diff
@@ -65,13 +71,22 @@ open class FullScreenFeature(
     override fun onBackPressed(): Boolean {
         val observation = observation
 
-        if (observation.inFullScreen && observation.tabId != null) {
+        if (!isFullscreenCustomTab && observation.inFullScreen && observation.tabId != null) {
             sessionUseCases.exitFullscreen(observation.tabId)
             return true
         }
 
         return false
     }
+
+    /**
+     * Method used to exit fullscreen mode, allowing fullscreen custom tabs to remain fullscreen.
+     */
+    fun exitFullscreenMode() {
+        if (!isFullscreenCustomTab) {
```

Comment:
❓ Is there a way to exit fullscreen custom tab then?

Reply:
For full-screen custom tabs, we only exit full-screen mode when the browser fragment is [destroyed](https://github.com/mozilla-mobile/firefox-android/pull/1258/files#diff-4385d548241ec2239de72ceb324eeb29f03cd502b71a29a1285be2f0f350c71dR1491). I added the call in `onDestroyView` as not to disrupt current `FullScreenFeature.onBackPressed()` [interactions](https://searchfox.org/mozilla-mobile/source/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt#1207). Should we allow the user to exit full-screen mode on back pressed for example?

Reply:
Huh, so if I open a youtube video in a custom tab, I can never leave it? 🙃

Reply:
I don't know how we could currently open a youtube video as directly fullscreen in a custom tab. I know we can use `youtube.com/watch_popup?v=` to open a video in fullscreen on dekstop but it doesn't work on mobile. 

In the case when we open a youtube link in a custom tab and then watch the video fullscreen, pressing back would call the `FullscreenFeature.onBackPressed()` with `isFullscreenCustomTab` set to `false` (the tab would be created as non-fullscreen). 

For the case when we open a fullscreen video in a fullscreen custom tab, I updated the first video from the PR description.

Should we maybe update the naming of the `isFullscreenCustomTab` flag or move the logic to the `BaseBrowserFragment` ?